### ISSUE TITLE
Support for Enum values in IConfiguration interface mapping to AWSOptions in AWSSDK.Extensions.NETCore.Setup 

### ISSUE THREAD

**Issue by madhub** [aws/aws-sdk-net#2587]
### Describe the feature

Currently Enum values such has , RetryMode mode is not supported during the mapping from IConfiguration to AWSOptions
Documentation does not list the enum configuration [items](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-netcore.html#Allowed%20values%20in%20appsettings%20file)
```json
appSettings.json
{
  "AWS": {
    "Profile": "local-test-profile",
    "Region": "us-west-2",
    "RetryMode": "Standard"
  }
}
```
```csharp
var options = Configuration.GetAWSOptions();
// options  will not have RetryMode eventhough it configured in 
IAmazonS3 client = options.CreateServiceClient<IAmazonS3>();
```

### Use Case

Developer shall be able to configure **RetryMode** via configuration file so that all service clients will use the same retry strategy and  developer shall be to configure different retry strategy  by changing the **configuration file** instead of **code**

### Proposed Solution

Proposed option is to add **new if/else condition to support Enum** in `GetAWSOptions(this IConfiguration config, string configSection)`
Proposed change is in [GetAWSOption function](https://github.com/sbuzonas/aws-sdk-net/blob/75c2e8986ba5be98c436d66cce2c6de671c1dcf3/extensions/src/AWSSDK.Extensions.NETCore.Setup/ConfigurationExtensions.cs#L85)

Updated GetAWSOptions function with Enum support 
```csharp
 public static AWSOptions GetAWSOptions(this IConfiguration config, string configSection)
        {
             ....
             ....
            var clientConfigTypeInfo = typeof(ClientConfig).GetTypeInfo();
            foreach(var element in section.GetChildren())
            {
                try
                {
                    var property = clientConfigTypeInfo.GetDeclaredProperty(element.Key);
                    if (property == null || property.SetMethod == null)
                        continue;

                    if (property.PropertyType == typeof(string) || property.PropertyType.GetTypeInfo().IsPrimitive)
                    {
                        var value = Convert.ChangeType(element.Value, property.PropertyType);
                        property.SetMethod.Invoke(options.DefaultClientConfig, new object[] { value });
                    }
                    else if (property.PropertyType == typeof(TimeSpan) || property.PropertyType == typeof(Nullable<TimeSpan>))
                    {
                        var milliSeconds = Convert.ToInt64(element.Value);
                        var timespan = TimeSpan.FromMilliseconds(milliSeconds);
                        property.SetMethod.Invoke(options.DefaultClientConfig, new object[] { timespan });
                    }
                     // code to support Enum values
                    else if (property.PropertyType.IsEnum) 
                    {
                        if ( Enum.TryParse(property.PropertyType, element.Value, out object propValue))
                        {
                               property.SetMethod!.Invoke(options.DefaultClientConfig, new object[1] { propValue });
                        }
                    }
                }
                catch(Exception e)
                {
                    throw new ConfigurationException($"Error reading value for property {element.Key}.", e)
                    {
                        PropertyName = element.Key,
                        PropertyValue = element.Value
                    };
                }
            }
          .....
          ....
}
```

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### AWS .NET SDK and/or Package version used

AWSSDK.S3 3.7.104.2

### Targeted .NET Platform

.NET 6

### Operating System and version

Windows,Linux

**Comment by ashishdhingra** [comment]
Hi @madhub,

Good morning.

Thanks for submitting feature request. At high level, it appears to be a valid feature request and the proposed solution should work. Feel free to contribute a PR that could be reviewed by the team.

Thanks,
Ashish

**Comment by madhub** [comment]
@ashishdhingra  this feature is merged to main #2592 . Do I need to close the issue myself or you will close.

**Comment by ashishdhingra** [comment]
> @ashishdhingra this feature is merged to main #2592 . Do I need to close the issue myself or you will close.

@madhub Thanks. AWSSDK.Extensions.NETCore.Setup v3.7.6 released.
